src/lib/helpers/getFetchClient.ts

```diff
@@ -1,18 +1,21 @@
 import { Store } from "@willsoto/node-konfig-core";
+import { retry, type RetryOptions } from "jsr:@std/async";
+
+type FetchInputParameter = Parameters<typeof fetch>[0];
+type FetchInitParameterWithClient =
+    | RequestInit
+    | RequestInit & { client: Deno.HttpClient };
+type FetchReturn = ReturnType<typeof fetch>;
 
 export const getFetchClient = (konfigStore: Store): {
-    (input: RequestInfo | URL, init?: RequestInit): Promise<Response>;
     (
-        input: Request | URL | string,
-        init?: RequestInit & {
-            client: Deno.HttpClient;
-        },
-    ): Promise<Response>;
-    (input: URL | Request | string, init?: RequestInit): Promise<Response>;
+        input: FetchInputParameter,
+        init?: FetchInitParameterWithClient,
+    ): FetchReturn;
```

Comment:
for reference, I've just replaced this section because I needed to use it below, so I've defined some types that cover the inputs to all functions.